src/lib/logger/Logger.ts

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export enum LogLevel {
+  test = 0,
+  trace = 1,
+  debug = 2,
+  info = 3,
+  warn = 4,
+  error = 5,
+  fatal = 6,
+  off = 7,
+}
+
+export interface ILogger {
+  logLevel: LogLevel;
+
+  test(message: string, data?: Record<string, any>): void;
+  trace(message: string, data?: Record<string, any>): void;
+  debug(message: string, data?: Record<string, any>): void;
+  info(message: string, data?: Record<string, any>): void;
+  warn(message: string, data?: Record<string, any>): void;
+  error(message: string, data?: Record<string, any>): void;
+  fatal(message: string, data?: Record<string, any>): void;
+}
```

Comment:
Mainly the interface of this file

Reply:
Good work Timo! It's almost that! I would recommend also to have a "per section" logger. This can help "activate/deactivate" loggers per section. Say:

```typescript

/**
 * We define the logger sections
 */
export enum LogSection {
  default = 'DEFAULT',
  transports = 'TRANSPORTS',
  whatever = 'WHATEVER',
}

export interface ILogger {
  logLevel: LogLevel;
  /**
   * All log instances will have a "section" value. If the default logger will fallback to "default"
   */
  readonly section: LogSection;

 // Other stuff

  /**
   * This method returns an instance of the logger, with the "section" set to the parameter.
   * If the section already exists, the same logger should be returned.
   *
   * Section loggers are disabled by default
   *
   * @param section the section name
   */
  logger(section: LogSection): ILogger;

  /**
   * Enable this section logger
   *
   * @param section the section name
   */
  enableLogger(section: LogSection): void;

  /**
   * Disable the section logger
   *
   * @param section the section name
   */
  disableLogger(section: LogSection): void;
}

// Then when you use the logger...

const aLogger: ILogger = ...;

const sectionLogger = aLogger.logger(LogSection.transports);
aLogger.enableLogger(LogSection.transports);
sectionLogger.debug('Whatever'); // This prints something like "SECTION: Whatever"
aLogger.disableLogger(LogSection.transports);
sectionLogger.debug('Whatever'); // This does not print anything

```

This can help you enable/disable the logs for some sections of the framework.

On the implementation, I would use transports as winston does. This can help enable file, NewRelic, JSON, Database or console transports with the same instance of the logger. Or even disable a transport for a level or section. But this is far away from "console" basic transport so there is no need to implement this. Anyone can do it if needed with this interface. I think the basic transport will do👌

Reply:
Thanks for the feedback @icc-romeu!

> On the implementation, I would use transports as winston does. This can help enable file, NewRelic, JSON, Database or console transports with the same instance of the logger. Or even disable a transport for a level or section. But this is far away from "console" basic transport so there is no need to implement this. Anyone can do it if needed with this interface. I think the basic transport will do👌



I started out with all these things but then realized I was creating a new logger library. I wanted to keep it as simple as possible to start with and then expand over time.

> Good work Timo! It's almost that! I would recommend also to have a "per section" logger. This can help "activate/deactivate" loggers per section. Say:

My idea was to allow to create scoped sub loggers with a name. The logger implementation can then handle all the specifics.